src/Kiota.Builder/Refiners/RubyRefiner.cs

```diff
@@ -0,0 +1,74 @@
+            ReplaceRelativeImportsByImportPath(generatedCode, '.');
+        }
+        private static readonly Tuple<string, string>[] defaultNamespacesForRequestBuilders = new Tuple<string, string>[] { 
+            new ("HttpCore", "microsoft_kiota_abstractions"),
+            new ("HttpMethod", "microsoft_kiota_abstractions"),
+            new ("RequestInfo", "microsoft_kiota_abstractions"),
+            new ("ResponseHandler", "microsoft_kiota_abstractions"),
+            new ("QueryParametersBase", "microsoft_kiota_abstractions"),
+            new ("SerializationWriterFactory", "microsoft_kiota_abstractions"),
+        };
+        private static readonly Tuple<string, string>[] defaultNamespaces = new Tuple<string, string>[] { 
+            new ("SerializationWriter", "microsoft_kiota_abstractions"),
+        };
+        private static readonly Tuple<string, string>[] defaultNamespacesForModels = new Tuple<string, string>[] { 
+            new ("ParseNode", "microsoft_kiota_abstractions"),
+            new ("Parsable", "microsoft_kiota_abstractions"),
+        };
+        private static readonly Tuple<string, string>[] defaultSymbolsForApiClient = new Tuple<string, string>[] { 
+            new ("ApiClientBuilder", "microsoft_kiota_abstractions"),
+            new ("SerializationWriterFactoryRegistry", "microsoft_kiota_abstractions"),
+            new ("ParseNodeFactoryRegistry", "microsoft_kiota_abstractions"),
+        };
+        private static void AddParsableInheritanceForModelClasses(CodeElement currentElement) {
+            if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model) 
+                && currentClass.StartBlock is CodeClass.Declaration declaration) {
+                declaration.Implements.Add(new CodeType(currentClass) {
+                    IsExternal = true,
+                    Name = $"MicrosoftKiotaAbstractions::Parsable",
+                });
+            }
+            CrawlTree(currentElement, AddParsableInheritanceForModelClasses);
+        }
+        protected static void AddInheritedAndMethodTypesImports(CodeElement currentElement) {
+            if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model) 
+                && currentClass.StartBlock is CodeClass.Declaration declaration && declaration.Inherits != null) {
+                currentClass.AddUsing(new CodeUsing(currentElement) { Name = declaration.Inherits.Name, Declaration = declaration.Inherits});
+            }
+            CrawlTree(currentElement, (x) => AddInheritedAndMethodTypesImports(x));
+        }
+
+        protected static void FixInheritedEntityType(CodeElement currentElement, CodeClass entityClass = null, string prefix = ""){
+            if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model) 
+                && currentClass.StartBlock is CodeClass.Declaration declaration && declaration?.Inherits != null 
```

Comment:
declaration will never be null, either it's of a valid type (and has a value) or it's null and the previous condition will return false. you can remove ? after delcaration

Reply:
```suggestion
                && currentClass.StartBlock is CodeClass.Declaration declaration && declaration.Inherits != null 
```